client/lib/protect-form/README.md

```diff
@@ -1,34 +1,73 @@
 Protect-Form
 ============
 
-This [Higher Order Component](https://medium.com/@dan_abramov/mixins-are-dead-long-live-higher-order-components-94a0d2f9e750) protects a user from navigating away from a form that has been edited without having been saved. When your form has been marked as changed, the browser will open a confirm dialog if you try to navigate away from the page without saving the changes.
+This component protects a user from navigating away from a form that has been edited without having been saved. When your form has been marked as changed, the browser will open a confirm dialog if you try to navigate away from the page without saving the changes.
 
-Usage
------
+The module exports two implementations. One is a
+[Higher Order Component](https://medium.com/@dan_abramov/mixins-are-dead-long-live-higher-order-components-94a0d2f9e750)
+that is used to wrap you form component.
 
-```es6
+The other is a declarative component that is supposed to be rendered inside the form. It
+has an API that is more suitable for integration with Redux state.
+
+Usage (HOC)
+-----------
+
+```js
 import { protectForm } from 'lib/protect-form';
 ```
 
 ... and then later in your component definition like so:
 
-```es6
+```js
 export default protectForm( MyComponent );
 ```
 
-The protect-form HoC provides a `markChanged` prop you can call on the `onChange` event of your form like this:
+The `protectForm` HoC provides a `markChanged` prop you can call on the `onChange` event of your form like this:
 
-```jsx
+```js
```

Comment:
GitHub does a better syntax highlighting for JSX when ` ```jsx` is used instead of ` ```js`, so I added it at the two places where it matters.